src/transaction/log_page_buffer.c

```diff
@@ -6177,6 +6179,14 @@ logpb_remove_archive_logs (THREAD_ENTRY * thread_p, const char *info_reason)
 
   last_deleted_arv_num--;
 
+  vacuum_first_pageid = vacuum_min_log_pageid_to_keep (thread_p);
```

Comment:
Could you let me know why _vacuum_is_safe_to_remove_archives()_ is not needed here? It is different from _logpb_remove_archive_logs_exceed_limit()_ .

Reply:
After vacuum_boot() is called, I think there is no problem with calling vacuum_min_log_pageid_to_keep(). However, for code consistency, I will call vacuum_is_safe_to_remove_archives() to make sure that the deletion of archive log volumes is safe.

Reply:
1. `logpb_remove_archive_logs_exceed_limit()` seems to be also called after vacuume_boot().
2. In my opinion, if it is missed in `logpb_remove_archive_logs()`. This function has to be always called when it is guaranteed that `vacuum_Data.is_archive_removal_safe` is set to true. It seems more stable to check it. Or you can `assert()` instead of my suggestion if the condition is pre-condition.

Reply:
* The logpb_remove_archive_logs_exceed_limit() function is called from logpb_archive_active_log().
* So it can be called even before vacuum_boot().
* Therefore, I think vacuum_is_safe_to_remove_archives() must be called in order to operate after vacuum_boot().
* However, the logpb_remove_archive_logs() function does not seem to be a problem because it is a function that is called at runtime after server boot. What do you think?

Reply:
How about replacing all codes (line 6172 ~ 6193)  that remove archive log to logpb_remove_archive_logs_exceed_limit() function ?
I think the function is safe, because it have routine such as checking  "force remove archive" option, "vacuum min log" and "system crash min log" before removing archive logs.

Reply:
@mhoh3963 Maybe it will work fine I think. However, running 'backupdb -r' will depend on the "log_max_archives" and "force_remove_log_archives" system properties. So, I think this is not a good choice. Nevertheless, if I need to replace it with the logpb_remove_archive_logs_exceed_limit() function, I think we need to reconsider the definition of the -r option. What do other reviewers(@JoohoK @wrlawodms) think?

Reply:
I think vacuum_is_safe_to_remove_archives() is better than prm_get_bool_value (PRM_ID_DISABLE_VACUUM) for code consistency and readability. So I used it. Please refer to the review. In addition, the vacuum_disable value is being checked by logpb_is_page_in_archive(), so I don't think any issues we discussed will occur. However, I want to call the function vacuum_is_safe_to_remove_archives() for explicit processing.